ostelco-ios-client/Extensions/UIWindow+ShakeToOpenHelp.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  UIWindow+ShakeToOpenHelp.swift
+//  ostelco-ios-client
+//
+//  Created by mac on 6/8/19.
+//  Copyright © 2019 mac. All rights reserved.
+//
+
+extension UIWindow {
+    open override func motionEnded(_ motion: UIEvent.EventSubtype, with event: UIEvent?) {
+        if motion == .motionShake {
+            guard let vc = UIApplication.shared.typedDelegate.rootCoordinator.topViewController else {
```

Comment:
I think you could do this with the window's `rootViewController` instead of needing to dig so deep into the properties of other objects.

Reply:
I tried with
`guard let vc = self.window?.rootViewController else {
                // There's no view controller to show.
                return
            }
`
without any luck

Reply:
In this case self *is* the window, so just `self.rootViewController` should be sufficient. Unless your issue was something other than a compiler problem?

Reply:
thanks! That worked